wfe/wfe.go

```diff
@@ -552,6 +552,78 @@ func (wfe *WebFrontEndImpl) verifyContacts(acct acme.Account) *acme.ProblemDetai
 	return nil
 }
 
+func (wfe *WebFrontEndImpl) UpdateAccount(
+	ctx context.Context,
+	logEvent *requestEvent,
+	response http.ResponseWriter,
+	request *http.Request) {
+	body, key, prob := wfe.verifyPOST(ctx, logEvent, request, wfe.lookupJWK)
+	if prob != nil {
+		wfe.sendError(prob, response)
+		return
+	}
+
+	// updateAcctReq is the ACME account information submitted by the client
+	var updateAcctReq struct {
+		Contact []string `json:"contact"`
+	}
+	err := json.Unmarshal(body, &updateAcctReq)
+	if err != nil {
+		wfe.sendError(
+			acme.MalformedProblem("Error unmarshaling body JSON"), response)
+		return
+	}
+
+	existingAcct, prob := wfe.getAcctByKey(key)
+	if prob != nil {
+		wfe.sendError(prob, response)
+		return
+	}
+
+	// if this update contains no contacts, simply return the existing account
+	// and return early.
+	if len(updateAcctReq.Contact) == 0 {
```

Comment:
@cpu if a user were to submit a request that explicitly states `contacts: []`, how should the ACME server behave? Should it throw an error because the user has requested to set no contacts, or should it assume this is a no-op (i.e. assume the user has submitted `{}` as the request body).

I think we can achieve this behaviour through use of `omitempty`, although this change in behaviour could break existing clients (as the spec does not seem to be explicit here).

Reply:
We handle this as a no-op in Boulder (equiv to submitting {}) and I think it's fair to keep that here:

https://github.com/letsencrypt/boulder/blob/2dc3b56fa9d68a935259bb199fc17523821cff04/ra/ra.go#L390:L392

Reply:
👍 that is the current behaviour, but just checking to make sure it's correct! Great 😄